README.md

```diff
@@ -0,0 +1,297 @@
+
+Where to Write Tests
+--------------------
+
+All of the OSG Software automated tests are located in the `osg-test` software and package.
+
+The software itself is in GitHub repository at <https://github.com/opensciencegrid/osg-test>; current code is kept in the `master` branch.
+
+The software package is defined in our Subversion repository at `native/redhat/trunk/osg-test`.
+
+### Directory Organization
+
+The test software is written in Python and consists of:
+
+-   A driver program, `osg-test`
+-   A set of support libraries (Python modules) in `osgtest/library`
+-   The tests themselves (also Python modules) in `osgtest/tests`
+-   Extra files needed at runtime in `files`
+
+The whole system uses the standard Python `unittest` framework to run. Note that all tests have to be compatible with Python 2.4; when reading the docs for `unittest`, keep note of when a feature was introduced.
+
+### Test Sequence
+
+During a test run, the test modules are run in sequence as follows:
+
+| File                 | When                                          | Purpose                                                               |
+|:---------------------|:----------------------------------------------|:----------------------------------------------------------------------|
+| `special_user.py`    | Tests not suppressed, or explicitly requested | Add user (if asked), Check user, Set up mapfile                       |
+| `special_install.py` | Packages given                                | Check repositories, Clean yum cache, Install packages                 |
+| `test_NN_*.py`       | Tests not suppressed                          | Configure, Test, Tear down                                            |
+| `special_cleanup.py` | Explicitly requested                          | Remove user (if added), Remove packages (if installed)                |
+
+The `test_*` modules are organized roughly into three phases, based on the sequence number of the file:
+
+| Test Files       | Purpose   |
+|:-----------------|:----------|
+| `test_[00-29]_*` | Set up    |
+| `test_[30-69]_*` | Tests     |
+| `test_[70-99]_*` | Tear down |
+
+Coding Tips
+-----------
+
+It is important to know the basics of the Python `unittest` module; [read the documentation for it](http://docs.python.org/release/2.4.3/lib/module-unittest.html). We build on top of the `unittest` module, by providing an `osgunittest` module that inherits from it.
+
+### Basic Structure of a Test Module
+
+Each test module must import the `osgunittest` library, plus whichever of the `osg-test` libraries are needed (conventionally with shortened aliases):
+
+```python
+import osgunittest
+
+import osgtest.library.core as core
+import osgtest.library.files as files
+```
+
+Then, a single test class is defined, deriving from =osgunittest.OSGTestCase=; the individual tests are sequentially numbered functions within the class:
```

Comment:
Replae `=` around `osgunittest.OSGTestCase` with backticks. (If anyone knows how to quote a backtick in markdown, tell me!)